base/Dockerfile.ubi8

```diff
@@ -42,30 +42,68 @@ ENV LANG="C.UTF-8"
+ARG IPUTILS_VERSION=""
+ARG HOSTNAME_VERSION=""
+
+# Zulu OpenJDK version
+ARG ZULU_OPENJDK_VERSION=""
+
+# Python Module Versions
+ARG PYTHON_PIP_VERSION=""
+ARG PYTHON_SETUPTOOLS_VERSION=""
 
-ENV PYTHON_VERSION="36-3.6.8"
+# Confluent Docker Utils Version (Namely the tag or branch to grab from git to install)
+ARG PYTHON_CONFLUENT_DOCKER_UTILS_VERSION="master"
 
-COPY requirements.txt .
+# This can be overriden for an offline/air-gapped builds
+ARG PYTHON_CONFLUENT_DOCKER_UTILS_INSTALL_SPEC="git+https://github.com/confluentinc/confluent-docker-utils@${PYTHON_CONFLUENT_DOCKER_UTILS_VERSION}"
 
 RUN microdnf install yum \
-    && yum update -y \
-    && yum install -y openssl git wget nc python${PYTHON_VERSION} tar procps krb5-workstation iputils hostname \
-    && alternatives --set python /usr/bin/python3 \
-    && python3 -m pip install --upgrade pip setuptools \
-    && python3 -m pip install --prefer-binary --prefix=/usr/local --upgrade 'git+https://github.com/confluentinc/confluent-docker-utils@v0.0.43' \
     && rpm --import http://repos.azulsystems.com/RPM-GPG-KEY-azulsystems \
-    && curl -o /etc/yum.repos.d/zulu.repo http://repos.azulsystems.com/rhel/zulu.repo \
-    && yum -y install ${ZULU_OPENJDK} \
+    && curl -fSL -o /etc/yum.repos.d/zulu.repo http://repos.azulsystems.com/rhel/zulu.repo \
+    && yum install -y --setopt=install_weak_deps=False \
+        git \
+        "openssl${OPENSSL_VERSION}" \
+        "wget${WGET_VERSION}" \
+        "nmap-ncat${NETCAT_VERSION}" \
+        "python36${PYTHON36_VERSION}" \
+        "tar${TAR_VERSION}" \
+        "procps-ng${PROCPS_VERSION}" \
+        "krb5-workstation${KRB5_WORKSTATION_VERSION}" \
+        "iputils${IPUTILS_VERSION}" \
+        "hostname${HOSTNAME_VERSION}" \
+        "zulu-8${ZULU_OPENJDK_VERSION}" \
+    && alternatives --set python /usr/bin/python3 \
+    && python3 -m pip install --upgrade "pip${PYTHON_PIP_VERSION}" "setuptools${PYTHON_SETUPTOOLS_VERSION}" \
+    && python3 -m pip install --prefer-binary --prefix=/usr/local --upgrade "${PYTHON_CONFLUENT_DOCKER_UTILS_INSTALL_SPEC}" \
     && yum remove -y git \
     && yum clean all \
     && rm -rf /tmp/* \
-    && rm -f requirements.txt \
     && mkdir -p /etc/confluent/docker /usr/logs \
     && useradd --no-log-init --create-home --shell /bin/bash appuser \
     && chown appuser:appuser -R /etc/confluent/ /usr/logs
 
+# This is a step that will cause the build to fail of the package manager detects a security update is availible and isn't installed.
+# The ARG DONT_FAIL_IF_SECURITY_UPDATE_AVAILIBLE is an "escape" hatch if you want to by-pass this check and build the container anyways, which
+# is not advisable in terms of security posture. If set to false (which triggers a shell exit(1) if the check fails from the ||) this check
+# will fail. If true (which triggers a right-hand || shell exit(0)), then this check will pass even if a security update is availible.
+ARG DONT_FAIL_IF_SECURITY_UPDATE_AVAILIBLE="false"
```

Comment:
TBH, the default value being double negative  (don't want it = false) is a bit hard to understand. How about FAIL_SECURITY_CHECK = true?

Reply:
I understand what you mean - but thats the tricky thing I'm doing with bash here - The value will be interpreted as `false` (exit 1) or `true` (exit 0) if the left hand of `||` fails. ie:

<check-fails> || true # Don't fail
<check-pass> || true # Don't fail
<check-fails> || false # Fail
<check-pass> || false # Don't fail (because the right hand of `||` is never evaluated)

Reply:
I renamed this to `docker.skip-security-update-check`